controllers/discovery/initDiscoveryDomainData.js

```diff
@@ -40,16 +46,30 @@ const InitDiscoveryDomainData = async (req, res) => {
       replacements: {
         userId,
         limit: limit,
-        offset: page * limit
+        offset: page * limit,
+        credderMinScore: CREDDER_MIN_SCORE
       }
     });
     let suggestedDomains = domainWithCommonFollowerResult;
 
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId,
+        credderMinScore: CREDDER_MIN_SCORE
+      },
+      raw: true
+    });
+    totalData = totalData?.[0]?.total_data || 0;
+
     return res.status(200).json({
       success: true,
       message: `Fetch discovery data success`,
       suggestedDomains,
-      page: page + 1
+      page: page + 1,
+      total_page: totalData > 0 && limit > 0 ? Math.ceil(totalData / limit) : 0,
+      limit: limit,
+      offset: page * limit
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are good for providing more information to the client, such as the total number of pages and the current offset. However, there's a potential issue with the calculation of `total_page`. If `limit` is 0, it will cause a division by zero error. To prevent this, you should add a check to ensure that `limit` is not zero before performing the division.
```diff
- total_page: totalData > 0 && limit > 0 ? Math.ceil(totalData / limit) : 0,
+ total_page: totalData > 0 && limit !== 0 ? Math.ceil(totalData / limit) : 0,
```
Also, consider improving error handling in the catch block. Currently, it only logs 'e' which doesn't provide much information about the error. It would be better to log the actual error object.
```diff
- console.log('e');
+ console.log(e);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->